### PR TITLE
Upgrade pants and toolchain pants plugin and rename lock file

### PR DESCRIPTION
**Pull request by asherf**

### CODE REVIEW

**Comment by Eric-Arellano** [comment]
Thanks! We need to set `use_deprecated_directory_cli_args_semantics = true`. I opened https://github.com/pantsbuild/example-python/issues/108 for me to update the README

File: pants.toml [hunk 1]
```diff
@@ -12,20 +12,31 @@ backend_packages.add = [
   "pants.backend.python.typecheck.mypy",
 ]
 
+use_deprecated_directory_cli_args_semantics = true
```
Comment (Eric-Arellano) [inline pants.toml:15]: Sorry, I gave you bad advice

```suggestion
use_deprecated_directory_cli_args_semantics = false
```

File: pants.toml [hunk 2]
```diff
@@ -12,20 +12,31 @@ backend_packages.add = [
...
 ]
 
 remote_cache_read = false
 remote_cache_write = false
 remote_store_address = "grpcs://cache.toolchain.com:443"
 remote_auth_plugin = "toolchain.pants.auth.plugin:toolchain_auth_plugin"
 
-# End Toolchain-specific configuration
+# Configuration for Toolchain instrumentation and remote caching
+# Visit https://toolchain.com and https://docs.toolchain.com/docs/getting-started-with-toolchain to enable and try it on your repository
```
Comment (asherf) [inline pants.toml:30]: @benjyw fyi

**Review by Eric-Arellano — APPROVED** [review]

**Comment by Eric-Arellano** [comment]
Hm the Pex binary changes also need to happen here. I'll push a change to this PR

**Pull request closed without merging**